From 1c0bdf44cb50ffee29af3cc4ceef15e8448a8d23 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Thu, 26 Apr 2018 23:33:37 +0800
Subject: [PATCH 3630/5242] MGS-3848-5 [#imx-854] OCL1.2: test_image_streams
 failures

commit  e781c63f8b57a9cdf97b45fbe8b11230206eeb90 from
https://source.codeaurora.org/external/imx/linux-imx.git

there are 2 test failed on 8QXP FB. CL151757 fixed bug #20196,
for image objects using host ptr, set the cacheable flag correctly.

merged CL151774 fix build error cause by CL151757.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c   |    1 -
 drivers/mxc/gpu-viv/hal/kernel/inc/gc_hal_enum.h   |    9 ++++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
index 92d43ca..03bd3c7 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
@@ -2217,7 +2217,6 @@
 #if gcdUSE_MMU_EXCEPTION
 #if gcdALLOC_ON_FAULT
             status = gckHARDWARE_HandleFault(Event->kernel->hardware);
-
 #endif
             if (gcmIS_ERROR(status))
             {
diff --git a/drivers/mxc/gpu-viv/hal/kernel/inc/gc_hal_enum.h b/drivers/mxc/gpu-viv/hal/kernel/inc/gc_hal_enum.h
index 983cd33..9a4328f 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/inc/gc_hal_enum.h
+++ b/drivers/mxc/gpu-viv/hal/kernel/inc/gc_hal_enum.h
@@ -1175,6 +1175,14 @@
 }
 gceSURF_FORMAT;
 
+typedef enum _gceIMAGE_MEM_TYPE
+{
+    gcvIMAGE_MEM_DEFAULT,
+    gcvIMAGE_MEM_HOST_PTR,
+    gcvIMAGE_MEM_HOST_PTR_UNCACHED,
+}
+gceIMAGE_MEM_TYPE;
+
 typedef enum _gceSURF_YUV_COLOR_SPACE
 {
     gcvSURF_ITU_REC601,
@@ -2168,4 +2176,3 @@
 #endif
 
 #endif /* __gc_hal_enum_h_ */
-
-- 
1.7.9.5

